pkg/lib/translation/service.go

```diff
@@ -70,30 +82,58 @@ func (s *Service) GetSenderForTestSMS(ctx context.Context) (sender string, err e
 }
 
 func (s *Service) emailMessageHeader(ctx context.Context, name SpecName, variables *PreparedTemplateVariables) (sender, replyTo, subject string, err error) {
-	t, err := s.translationMap(ctx)
+	baseTranslations, err := s.translationMap(ctx)
 	if err != nil {
 		return
 	}
 
-	sender, err = t.RenderText(fmt.Sprintf("email.%s.sender", name), variables)
-	if errors.Is(err, template.ErrNotFound) {
-		sender, err = t.RenderText("email.default.sender", variables)
+	resolveSender := func(t *template.TranslationMap) (string, error) {
+		sender, err := t.RenderText(fmt.Sprintf("email.%s.sender", name), variables)
+		if errors.Is(err, template.ErrNotFound) {
+			sender, err = t.RenderText("email.default.sender", variables)
+		}
+		if err != nil {
+			return "", err
+		}
+		return sender, nil
 	}
-	if err != nil {
-		return
+
+	// Resolve sender
+	// If no smtp secret, probably in local, just use sender in translation
+	if s.SMTPServerCredentialsSecretItem == nil {
+		sender, err = resolveSender(baseTranslations)
+		if err != nil {
+			return
+		}
+	} else {
+		// If the secret has sender, use it
+		if s.SMTPServerCredentialsSecretItem.GetData().Sender != "" {
+			sender = s.SMTPServerCredentialsSecretItem.GetData().Sender
+		} else {
+			// Depends on the secret fs level, resolve sender from different level of translation
+			var levelTranslations *template.TranslationMap
+			levelTranslations, err = s.levelSpecificTranslationMap(ctx, s.SMTPServerCredentialsSecretItem.FsLevel)
+			if err != nil {
+				return
+			}
+			sender, err = resolveSender(levelTranslations)
+			if err != nil {
+				return
+			}
+		}
 	}
 
-	replyTo, err = t.RenderText(fmt.Sprintf("email.%s.reply-to", name), variables)
+	replyTo, err = baseTranslations.RenderText(fmt.Sprintf("email.%s.reply-to", name), variables)
 	if errors.Is(err, template.ErrNotFound) {
-		replyTo, err = t.RenderText("email.default.reply-to", variables)
+		replyTo, err = baseTranslations.RenderText("email.default.reply-to", variables)
```

Comment:
We overlooked this. `Reply-To: ` should have the same treatment with `From: `

I also found that Reply-To in default translation is empty, but in infra/mail/sender.go, we did not check Reply-To is empty or not and always include it. I think it does not seem correct.

In summary, From and Reply-To should always be used with the corresponding level of SMTP secret. The normal resolution logic applies to Subject.